File: 404. Sum of Left Leaves/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var sumOfLeftLeaves = function(root) {
  let res = 0;
  if (root) {
    if (root.left && !root.left.left && !root.left.right) res += root.left.val;
    return res + sumOfLeftLeaves(root.left) + sumOfLeftLeaves(root.right);
  }
  return res;
};

File: 404. Sum of Left Leaves/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        if not root or (not root.left and not root.right):
            return 0
        res = 0
        q = [root]
        while q:
            t = q.pop(0)
            if t.left and not t.left.left and not t.left.right:
                res += t.left.val
            if t.left:
                q.append(t.left)
            if t.right:
                q.append(t.right)
        return res
